framework/src/mesh/GeneratedMesh.C

```diff
@@ -43,6 +43,8 @@ InputParameters validParams<GeneratedMesh>()
   params.addParam<Real>("zmax", 1.0, "Upper Z Coordinate of the generated mesh");
   params.addParam<MooseEnum>("elem_type", elem_types, "The type of element from libMesh to generate (default: linear element for requested dimension)");
 
+  params.addParam<std::vector<SubdomainID> >("subdomain_ids", "Subdomain IDs of all generated elements (refer to libMesh::MeshTools::Generation for element ordering");
+
```

Comment:
Phasefield users use Generated Mesh almost exclusively for production runs. Yes the could simply ignore this parameter, but I just want to make sure it belongs here in one of the base classes of MOOSE. We could put this parameter in a different group.